File: backend-ebanking/src/main/java/com/bank/ebanking/model/Utilisateur.java
package com.bank.ebanking.model;

import lombok.Data;

import javax.persistence.*;

@Entity
@Inheritance(strategy = InheritanceType.JOINED)
public @Data  class Utilisateur {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;
    String nom;
    String prenom;
    String cin;
    String adresse;
    String telephone;
    String email;
    @Column(unique=true,nullable=false)
    String username;
    @Column(nullable=false)
    String password;
    @Column(nullable=false)
    String role;
}

File: frontend-ebanking/src/app/interfaces/utilisateur.ts
export class User {
    id: number;
    nom: string;
    prenom: string;
    cin: string;
    adresse: string;
    telephone: string;
    email: string;
    username: string;
    password: string;
    role: string;
    constructor(username:string,email:string,nom:string,prenom:string,adresse:string,password:string,cin:string,telephone:string){
            this.username = username;
            this.email = email;
            this.nom = nom;
            this.prenom = prenom;
            this.adresse = adresse;
            this.password = password;
            this.cin = cin;
            this.telephone = telephone; 
    }
}
